src/startUp.ts

```diff
@@ -103,11 +135,13 @@ function waitForStop(service: PisaService) {
         // unless an error or process.exit() happens)
         stdin.resume();
         stdin.setEncoding("utf8");
-        stdin.on("data", key => {
+        stdin.on("data", async key => {
             // ctrl-c ( end of text )
             if (key === "\u0003") {
                 // stop the pisa service
-                service.stop();
+                await service.stop();
+                // shut the db
+                await db.close();
```

Comment:
You could increase parallelism:
```
await Promise.all([
  service.stop(),
  db.close()
]);
```
Although it's slightly less readable.

Reply:
updated